[18.0][MIG] account_reconcile_model_oca: Migration to 18.0

xaviedoanhduy opened the pull request:
### change in v18

- `get_sql` has removed in https://github.com/odoo/odoo/commit/5c397d56eb33dc37a0b0a8d1e7d4c43583c95950
- `account_journal_payment_credit_account_id` has removed in https://github.com/odoo/odoo/commit/01b87f1230beac0568f4e3b1b76e547909506892

etobella committed:
[ADD] account_reconcile_model_oca

Commit:
[UPD] Update account_reconcile_model_oca.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (2 of 2 strings)

Translation: account-reconcile-17.0/account-reconcile-17.0-account_reconcile_model_oca
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-17-0/account-reconcile-17-0-account_reconcile_model_oca/it/

pedrobaeza committed:
[FIX] *: pre-commit fixes

Due to latest copier template with all the checks.

OCA-git-bot committed:
[BOT] post-merge updates

xtanuiha committed:
Added translation using Weblate (Chinese (Simplified) (zh_CN))

xtanuiha committed:
Translated using Weblate (Chinese (Simplified) (zh_CN))

Currently translated at 100.0% (2 of 2 strings)

Translation: account-reconcile-17.0/account-reconcile-17.0-account_reconcile_model_oca
Translate-URL: https://translation.odoo-community.org/projects/account-reconcile-17-0/account-reconcile-17-0-account_reconcile_model_oca/zh_CN/

xaviedoanhduy committed:
[IMP] account_reconcile_model_oca: pre-commit auto fixes

xaviedoanhduy committed:
[MIG] account_reconcile_model_oca: Migration to 18.0

thienvh332 commented on account_reconcile_model_oca/models/account_bank_statement_line.py:
```diff
@@ -0,0 +1,130 @@
+                    LIMIT 1
+                """,
+                params + [self.company_id.id],
+            )
```
Should use odoo.tools.SQL to avoid sql problems

thienvh332 reviewed (commented):
LG overall,

thienvh332 reviewed (approved):
You should note the problem with the `account_accountant_batch_payment` module in the description to avoid confusing reviewers.

nilshamerlinck commented:
can you tell more about that `account_accountant_batch_payment` incompatibility?

xaviedoanhduy commented:
> can you tell more about that `account_accountant_batch_payment` incompatibility?

The `account_accountant` module in enterprise has overridden the `_get_invoice_matching_st_line_tokens` function and returns a tuple of 3 elements ([detail](https://github.com/odoo/enterprise/blob/971e225c85508edc5baa0735e79b278fbf57703b/account_accountant/models/account_reconcile_model.py#L166)) , but with the `_get_invoice_matching_st_line_tokens` OCA function, it only returns a list ([detail](https://github.com/xaviedoanhduy/account-reconcile/blob/c4d1fadfc7a541f71f4d723a442226e1711cc7c4/account_reconcile_model_oca/models/account_reconcile_model.py#L343)). The incompatibility is that the return results of the 2 overridden functions are different.

thienvh332 commented:
Because in the enterprise accounting module system, there is `auto_install = True`. So if your source code has it, it will be automatically installed. And because the 2 modules **`account_accountant_batch_payment`** and **`sale_account_accountant`** have overwrites that overlap with this module, it leads to an error.

nilshamerlinck commented:
thanks for the information

then can you please prepare an additional PR on v17.0 to add a new `excludes:"account_accountant"` in the manifest of this module

in the PR description, please explain why, something like:

This code of `account_reconcile_model_oca` was originally extracted from Odoo CE 16.0. Meanwhile the `account.reconcile.model` model evolved on its side in Odoo EE, with impactful methods signatures changes. As a consequence `account_reconcile_model_oca` is incompatible with Odoo EE.

and then include commit of that PR in your 18.0 migration PR

xaviedoanhduy committed:
[IMP] account_reconcile_model_oca: excludes account_accountant module

victoralmau committed:
[IMP] account_reconcile_model_oca: Create _get_write_off_move_move_line_dict() method to allow extension

FWP from 16.0: https://github.com/odoo/odoo/pull/188808

Related to OCA/account-reconcile#761

TT51885

OCA-git-bot committed:
[IMP] update dotfiles
